one_shot_scripts/update_deck_format.js

```diff
@@ -0,0 +1,332 @@
+
+  cards = cards.map((id) => {
+    if (Array.isArray(id)) {
+      if (id.length <= 0) {
+        const details = carddb.getPlaceholderCard('');
+        return {
+          tags: [],
+          colors: details.colors,
+          cardID: details._id,
+          cmc: details.cmc,
+          type_line: details.type,
+          details: details,
+        };
+      }
+      if (id[0].cardID) {
+        id = id[0].cardID;
+      } else {
+        id = id[0];
+      }
+    } else if (id.cardID) {
+      id = id.cardID;
+    }
+    const details = carddb.cardFromId(id);
+    return {
+      tags: [],
+      colors: details.colors,
+      cardID: details._id,
+      cmc: details.cmc,
+      type_line: details.type,
+      details: details,
+    };
+  });
+
+  const elos = await cubefn.getElo(cards.map((card) => card.details.name));
+  const nonlands = cards.filter((card) => !card.type_line.toLowerCase().includes('land'));
+  const lands = cards.filter((card) => card.type_line.toLowerCase().includes('land'));
+
+  sort_fn = function(a, b) {
+    if (bot) {
+      return botRating(b, bot, elos[b.details.name]) - botRating(a, bot, elos[a.details.name]);
+    } else {
+      return (elos[b.details.name] = elos[a.details.name]);
+    }
+  };
+
+  nonlands.sort(sort_fn);
+  lands.sort(sort_fn);
+
+  const main = nonlands.slice(0, 23).concat(lands.slice(0, 17));
+  const side = nonlands.slice(23).concat(lands.slice(17));
+
+  const deck = [];
+  const sideboard = [];
+  for (let i = 0; i < 16; i += 1) {
+    deck.push([]);
+    if (i < 8) {
+      sideboard.push([]);
+    }
+  }
+
+  for (const card of main) {
+    let index = Math.min(card.cmc, 7);
+    if (!card.type_line.toLowerCase().includes('creature')) {
+      index += 8;
+    }
+    deck[index].push(card);
+  }
+  for (const card of side) {
+    sideboard[Math.min(card.cmc, 7)].push(card);
+  }
+
+  console.log(sideboard);
+  return {
+    deck,
+    sideboard,
+  };
+}
+
+async function update(deck) {
+  //has old format to update
+  if (deck.owner && deck.owner.length > 0) {
+    const draft = deck.draft ? await Draft.findById(deck.draft) : null;
+
+    if (
+      deck.newformat == false &&
+      deck.cards[deck.cards.length - 1] &&
+      typeof deck.cards[deck.cards.length - 1][0] === 'object'
+    ) {
+      //old format
+      deck.seats = [];
+
+      const playerdeck = await buildDeck(deck.cards[0]);
+
+      const playerSeat = {
+        bot: null,
+        userid: deck.owner,
+        username: deck.username,
+        pickorder: deck.cards[0],
+        name: deck.name,
+        description: deck.description,
+        cols: 16,
+        deck: playerdeck.deck,
+        sideboard: playerdeck.sideboard,
+      };
+
+      deck.seats.push(playerSeat);
+
+      //add bots
+      for (let i = 1; i < deck.cards.length; i += 1) {
+        //need to build a deck with this pool...
+        const botdeck = await buildDeck(deck.cards[i]);
+        const bot = {
+          bot: deck.bots[i - 1],
```

Comment:
There is a bot for every card in the deck?

Reply:
deck.bots is where the old bot data is stored. it was an array of size equal to seats - 1, so 7 for an 8 person table. we need to pull this data into the seat object, which is why we do it like this